packages/babel-core/test/config-ts.js

```diff
@@ -0,0 +1,82 @@
+import { loadPartialConfigSync } from "../lib/index.js";
+import path from "path";
+import { fileURLToPath } from "url";
+import { createRequire } from "module";
+import semver from "semver";
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const require = createRequire(import.meta.url);
+
+// We skip older versions of node testing for two reasons.
+// 1. ts-node does not support the old version of node.
+// 2. In the old version of node, jest has been registered in `require.extensions`, which will cause babel to disable the transforming as expected.
+
+describe("@babel/core config with ts [dummy]", () => {
+  it("dummy", () => {
+    expect(1).toBe(1);
+  });
+});
+
+semver.gte(process.version, "12.0.0")
+  ? describe
+  : describe.skip("@babel/core config with ts", () => {
+      it("should work with simple .cts", () => {
+        const config = loadPartialConfigSync({
+          configFile: path.join(
+            __dirname,
+            "fixtures/config-ts/simple-cts/babel.config.cts",
+          ),
+        });
+
+        expect(config.options.targets).toMatchInlineSnapshot(`
+      Object {
+        "node": "12.0.0",
+      }
+    `);
+      });
+
+      it("should throw with invalid .ts register", () => {
```

Comment:
Can you add a new test case for invalid `.cts` register? Since that is how we are detecting ts support.

Reply:
There is actually a special situation here. I found that `ts-node` will only register `.ts`, and other extensions are realized by hijacking `.js`.
I seem to need to add some comments.
Actually I think the best way to detect it here is probably to directly try to `require()` a file with only `type xxx`, but that would require us to include it in the package, I'm not sure it's worth it. (Also future `.cts .ts .mts` may require three different files, which is kind of ugly)